internal/objectstore/fileobjectstore.go

```diff
@@ -0,0 +1,424 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package objectstore
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/juju/errors"
+	"gopkg.in/tomb.v2"
+
+	"github.com/juju/juju/core/objectstore"
+)
+
+type opType int
+
+const (
+	opGet opType = iota
+	opPut
+	opRemove
+)
+
+type request struct {
+	op            opType
+	path          string
+	reader        io.Reader
+	size          int64
```

Comment:
Nit: uint64?

Reply:
The ship has sailed on that one. I'd have to change everywhere. For this instance, I think it should be fine.